### PR TITLE
ci: Benchmark with codspeed

### PR DESCRIPTION
**Pull request by sdankel**
## Description

Closes https://github.com/FuelLabs/sway/issues/6477

## Checklist

- [ ] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
   - [ ] If my change requires substantial documentation changes, I have [requested support from the DevRel team](https://github.com/FuelLabs/devrel-requests/issues/new/choose)
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Review by kayagokalp — APPROVED** [review]

**Review by JoshuaBatty — CHANGES_REQUESTED** [review]

File: .github/workflows/bench-lsp.yml [hunk 1]
```diff
@@ -1,26 +0,0 @@
-name: Benchmark LSP
-
-on:
-  pull_request:
-    draft: false
-    paths:
-      - forc-plugins/forc-lsp/**
-      - sway-lsp/**
-      - forc-pkg/**
-      - sway-core/**
```
[the remark this answers is not in the document]
Comment (sdankel) [reply .github/workflows/bench-lsp.yml:10]: I believe it's meant to run on merges to master to create the new baseline for comparison.

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: Codspeed Benchmarks
+
+on:
+  push:
+    branches:
+      - "master"
+  pull_request:
```
Comment (sdankel) [inline .github/workflows/benchmark.yml:7]:
```suggestion
  pull_request:
    paths: 
       - forc-plugins/forc-lsp/** 
       - sway-lsp/** 
       - forc-pkg/** 
       - sway-core/**
```

**Comment by kayagokalp** [comment]
In a separate PR, maybe we should also get sway compiler benchmarks to this system as well, rather than collecting them in a separate website (which I believe happens right now). This way the compiler team will also be able to see their PR's effects more easily.

**Comment by sdankel** [comment]
> In a separate PR, maybe we should also get sway compiler benchmarks to this system as well, rather than collecting them in a separate website (which I believe happens right now). This way the compiler team will also be able to see their PR's effects more easily.
> 
> I guess we should first experiment with our own benchmarks and can recommend them if it looks good

This will run and upload all criterion benchmarks to codspeed. I believe the compiler benchmarks aren't run with criterion because they're measuring things other than speed (memory, bytecode size, etc)

**Comment by sdankel** [comment]
Closing in favor of https://github.com/FuelLabs/sway/pull/6481

I'm not sure why, but it randomly started failing on this PR and works on the new branch with the same changes.

**Pull request closed without merging**